### PR TITLE
Revert "Http client/server fixes"

### PR DESCRIPTION
**Pull request by cmonr**
Reverts armPelionEdge/maestro#16

> [22406.467527] Alignment trap: node (8140) PC=0x7481be8c Instr=0xe8e52302 Address=0x01c0c6b6 FSR 0x801
[22406.467532] Alignment trap: node (8140) PC=0x7481be8c Instr=0xe8e52302 Address=0x01c0c6be FSR

Found by locally testing and running on a RPi.

### CODE REVIEW

File: httpSymphonyClient.go [hunk 1]
```diff
@@ -252,11 +252,10 @@ func (client *Client) startTicker() {
 // the client worker goroutine
 // does the sending of data to the server
 func (client *Client) clientWorker() {
-	closeHttp := func(r *http.Response) {
+	closeit := func(r *http.Response, buf *logBuffer) {
 		r.Body.Close()
-	}
-	closeBuf := func(buf *logBuffer) {
 		greasego.RetireCallbackData(buf.data)
+		debugging.DEBUG_OUT("  OKOKOKOKOKOKOKOKOK -----> retired callback data\n\n")
```
Comment (SenRamakri) [inline httpSymphonyClient.go]: Can we clean up these DEBUG_OUT msgs instead of saying OKOK....

File: httpSymphonyClient.go [hunk 2]
```diff
@@ -274,28 +273,23 @@ func (client *Client) clientWorker() {
...
+		//	    req.Header.Set("X-Custom-Header", "myvalue")
+		req.Header.Set("Content-Type", "application/json")
+		// req.Header.Add("X-Symphony-ClientId", client.clientId)
+
+		// client := &http.Client{}
+		resp, err := client.httpClient.Do(req)
 		if err != nil {
-			debugging.DEBUG_OUT("XXXXXXXXXXXXXXXXXXXXXXX error on new request %+v\n", err)
-			closeBuf(next)
+			debugging.DEBUG_OUT("XXXXXXXXXXXXXXXXXXXXXXX error on sending request %+v\n", err)
```
Comment (SenRamakri) [inline httpSymphonyClient.go]: Please remove XXXXXXXXX....

File: wwrmi/api.go [hunk 3]
```diff
@@ -838,10 +836,11 @@ func (client *Client) postLogs() (err error) {
 
 	if err == nil {
 		resp, err = client.client.Do(req)
-		// close response body in case of success and errors, not required if no response
-		if resp != nil {
-                        defer resp.Body.Close()
-                }
+		if err != nil {
```
Comment (SenRamakri) [inline wwrmi/api.go]: Should we collapse these nested double checks? Are we not supposed to Close if err != Nil.
Also if we have an error can we add an error msg/print.

File: wwrmi/statssender.go [hunk 4]
```diff
@@ -340,8 +340,10 @@ func (sndr *statSender) postStats() (err error) {
 
 	if err == nil {
 		resp, err = sndr.client.client.Do(req)
-		if resp != nil {
-			defer resp.Body.Close()
+		if err != nil {
```
Comment (SenRamakri) [inline wwrmi/statssender.go]: Should we collapse here as well or add an error print when err != nil ?

**Comment by cmonr** [comment]
@SenRamakri This was a pure revert PR for #16, because local testing that @deepikabhavnani was doing found that maestro wasn't actually connecting. 

The logic will likely need to be fixed, but for now, the debug statements can be fixed/removde a bit later.

**Comment by SenRamakri** [comment]
@cmonr - I didn't realize it until later. No worries.

File: httpSymphonyClient.go (hunk 1 above)
Comment (deepikabhavnani) [reply httpSymphonyClient.go]: I am not cleaning any of the debug messages added by Ed, will pick that cleanup once the functionality is done and bugs are fixes

File: wwrmi/api.go (hunk 3 above)
Comment (deepikabhavnani) [reply wwrmi/api.go]: No err and resp are separate. we can have non nil response even in case of error

**Pull request merged**